elastic/kibana
# [Cases] Hide cases in stack management UI

## Summary

fixes https://github.com/elastic/kibana/issues/160337

This PR 
- hides cases in the serverless  Elasticsearch project, cases APIs throw error 
- throws 403 from API when `owner=cases` for security or observability serverless mode
- verifies the behaviour in serverless functional as well as api_integration tests

**How to test**

- Boot up `es` serverless solution and make sure that `cases` from the navbar is hidden and cannot not be accessible through url  as well
- Boot up `observability` or `security` serverless solutions and make sure that `cases` is available in the navbar and works fine
- Boot up classic kibana and make sure that the left navbar has the same menu entries it always had.

### Checklist

Delete any items that are not applicable to this PR.



https://github.com/elastic/kibana/assets/117571355/20c1974e-44f0-45b0-80aa-e644fec148ff

### For maintainers
